### PR TITLE
Enhancement/add interceptors and cookies

### PR DESCRIPTION
**Pull request by amr-hany-ib**
- Added optional cookies and custom authenticator/interceptor
- Added chucker

### CODE REVIEW

**Review by FadyRaafat — APPROVED** [review]

**Review by Andrew2120 — APPROVED** [review]

File: BlabsNetwork/src/main/java/com/blabs/blabsnetwork/delegate/NetworkRequestDelegate.kt [hunk 1]
```diff
@@ -145,13 +166,15 @@ class NetworkRequestDelegate(
...
-                                            responseBody, object : TypeToken<E>() {}.type
+                                            responseBody,
+                                            object : TypeToken<E>() {}.type
                                         )
                                         NetworkResponse.Error(
                                             NetworkError.CustomServerError(response.code),
                                             errorModel
                                         )
                                     } catch (e: JsonSyntaxException) {
+                                        Log.e("NetworkRequestDelegate", e.stackTraceToString())
```
Comment (zerox321) [inline BlabsNetwork/src/main/java/com/blabs/blabsnetwork/delegate/NetworkRequestDelegate.kt]: please remove logger @amr-hany-ib

File: BlabsNetwork/src/main/java/com/blabs/blabsnetwork/delegate/NetworkRequestDelegate.kt [hunk 2]
```diff
@@ -186,25 +209,22 @@ class NetworkRequestDelegate(
                                  */
                                 try {
                                     val result = gson.fromJson<T>(
-                                        responseBody, object : TypeToken<T>() {}.type
+                                        responseBody,
+                                        object : TypeToken<T>() {}.type
                                     )
                                     continuation.resume(NetworkResponse.Success(result))
                                 } catch (e: JsonSyntaxException) {
+                                    Log.e("NetworkRequestDelegate", e.stackTraceToString())
```
Comment (zerox321) [inline BlabsNetwork/src/main/java/com/blabs/blabsnetwork/delegate/NetworkRequestDelegate.kt]: please remove logger @amr-hany-ib

File: BlabsNetwork/src/main/java/com/blabs/blabsnetwork/delegate/NetworkRequestDelegate.kt [hunk 3]
```diff
@@ -1,25 +1,31 @@
 package com.blabs.blabsnetwork.delegate
 
+import android.content.Context
+import android.util.Log
```
Comment (zerox321) [inline BlabsNetwork/src/main/java/com/blabs/blabsnetwork/delegate/NetworkRequestDelegate.kt]: remove logger import @amr-hany-ib

**Review by zerox321 — CHANGES_REQUESTED** [review]
please remove log .e messages in NetworkRequestDelegate method

**Review by zerox321 — APPROVED** [review]

**Pull request merged**